Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -214,6 +242,105 @@ noncomputable def powerBasisOfFiniteOfSeparable : PowerBasis F E :=
 
 end SeparableAssumption
 
+section FiniteIntermediateField
+
+-- TODO: generalize to 0 ≤ m ≠ n; show [F⟮α⟯ : F⟮α ^ m⟯] = m when m > 0.
+theorem isAlgebraic_of_adjoin_eq_adjoin {α : E} {m n : ℕ} (hm : 0 < m) (hmn : m < n)
+    (heq : F⟮α ^ m⟯ = F⟮α ^ n⟯) : IsAlgebraic F α := by
```

Comment:
Thanks for taking up my challenge. The proofs you come up with are nice and direct. The proof of `finiteDimensional_of_finite_intermediateField` was unexpected to me: the proof I have in mind is to keep adjoining elements to build an infinite chain of intermediate fields.

However, the statement of this theorem is still a bit sub-mathlib-standard, but I don't think it's worth spending much more efforts on it, so I left a to-do item as a comment.

Reply:
Oh, I haven't think of your generalizations before. It looks cool. 😄

Reply:
Do you mean [this](https://github.com/leanprover-community/mathlib4/pull/7788/commits/03ea98ffef8ddb7136916992559b42bd626f7d11)? Sorry for the oversight, I was operating in proof-by-contradiction mode.

Reply:
`[F⟮X⟯ : F⟮X^m⟯] = m` corresponds to the fact that `X -> X^m` is a degree m self-(branched)-cover of P^1.
To show the equality you mostly just need to prove the polynomial `X^m-Y : F⟮Y⟯[X]` is irreducible. Since `F⟮Y⟯` is the fraction field of `F[Y]` (which is integrally closed), by [Polynomial.Monic.irreducible_iff_irreducible_map_fraction_map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Polynomial/GaussLemma.html#Polynomial.Monic.irreducible_iff_irreducible_map_fraction_map) you reduce to showing it's irreducible in F[Y][X], where you can use [Polynomial.irreducible_of_eisenstein_criterion](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/EisensteinCriterion.html#Polynomial.irreducible_of_eisenstein_criterion). You will need to develop some API around `transcendental`, e.g. show that F⟮α⟯ is isomorphic to F⟮Y⟯=Frac F[Y] when α is transcendental.

Reply:
A few more missing APIs:

+ An AlgHom between two extensions induces an OrderEmbedding between the IntermediateField types (in particular `IntermediateField.lift` is injective). An AlgEquiv induces an OrderEquiv.

+ A transcendental element α over F induces an AlgEquiv between F⟮α⟯ and [RatFunc](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/RatFunc.html#RatFunc) F.

Then you can just show that `RatFunc F` over `F` has infinitely many intermediate fields. But at some point you still need to identify the intermediate field `F⟮X^m⟯` with `RatFunc F` itself ...